core/src/thread_parker/linux.rs

```diff
@@ -75,6 +93,12 @@ impl ThreadParker {
             let diff = timeout - now;
             if diff.as_secs() as libc::time_t as u64 != diff.as_secs() {
                 // Timeout overflowed, just sleep indefinitely
+                // REVIEW: elsewhere in libstd when we encounter this situation
+                // we simply loop until the timeout elapses, could that be done
+                // here instead of parking indefinitely? It's a bit of a moot
+                // point in the sense that indefinitely vs sleeping for years
+                // isn't really that different, but it's probably good to be
+                // consistent.
```

Comment:
I disagree. I think that sleeping indefinitely is the better solution since it won't consume 100% CPU.

Reply:
My point isn't consuming 100% cpu, it's more like if you ask to sleep for 1000 years and the system can only sleep for 800 years, we sleep for 800 and then we sleep for 200.